pydantic/_internal/_fields.py

```diff
@@ -326,20 +326,17 @@ def collect_model_fields(  # noqa: C901
                 # Note that we only do this for method descriptors for now, we might want to
                 # extend this to any descriptor in the future (by simply checking for
                 # `hasattr(assigned_value.default, '__get__')`).
-                assigned_value.default = assigned_value.default.__get__(None, cls)
-
-            # The `from_annotated_attribute()` call below mutates the assigned `Field()`, so make a copy:
-            original_assignment = (
-                copy(assigned_value) if not evaluated and isinstance(assigned_value, FieldInfo_) else assigned_value
-            )
+                default = assigned_value.default.__get__(None, cls)
+                assigned_value.default = default
+                assigned_value._attributes_set['default'] = default
```

Comment:
Next step would be to get rid of the `_attributes_set` logic when merging instances, which is error prone as you need to update it whenever you do a manual assignment on a `FieldInfo` instance.